utils/token/accesscontroller.go

```diff
@@ -0,0 +1,215 @@
+package token
```

Comment:
I'm actually going to try and upstream these auth pieces back into distribution. I don't particularly want a forked version I have to keep up to date if somebody changes something. The version in docker/distribution is basically the canonical version for Docker but it was too tightly coupled with net/http and I needed to make progress.